frontend/src/metabase/admin/performance/components/StrategyForm.tsx

```diff
@@ -94,14 +94,34 @@ export const StrategyForm = ({
   buttonLabels?: ButtonLabels;
   isInSidebar?: boolean;
 }) => {
-  const defaultStrategy: CacheStrategy = {
-    type: targetId === rootId ? "nocache" : "inherit",
-  };
+  const defaultStrategy: CacheStrategy = useMemo(
+    () => ({
+      type: targetId === rootId ? "nocache" : "inherit",
+    }),
+    [targetId],
+  );
+
+  // Remove default values from the form
+  const initialValues = useMemo(() => {
+    const strategy: any = { ...(savedStrategy ?? defaultStrategy) };
+    for (const fieldName in strategy) {
+      const value = strategy[fieldName as keyof CacheStrategy];
+      const defaultValue = getDefaultValueForField(strategy.type, fieldName);
+      if (
+        defaultValue !== undefined &&
+        defaultValue !== null &&
+        value === defaultValue
+      ) {
+        delete strategy[fieldName as keyof CacheStrategy];
+      }
+    }
+    return strategy;
+  }, [savedStrategy, defaultStrategy]);
```

Comment:
```suggestion
  // Remove default values from the form
  const initialValues = useMemo(() => {
    return _.chain(savedStrategy ?? defaultStrategy)
      .pairs()
      .filter(([fieldName, value]) => {
        const defaultValue = getDefaultValueForField(strategy.type, fieldName);
        return defaultValue !== undefined && defaultValue !== null && value === defaultValue;
      })
      .object()
      .value();
  }, [savedStrategy, defaultStrategy]);
```

^ something like this reads a little cleaner IMO

Reply:
Thanks! I used more underscore in the new draft